posthog/models/feature_flag/flag_matching.py

```diff
@@ -483,78 +520,41 @@ def get_all_feature_flags(
-            # since the set_feature_flag_hash_key_overrides call will fail.
-
-            # For this case, and for any other case, do not error out on decide, we can handle it!
-            capture_exception(e)
-
-    # :TRICKY: Consistency matters only when personIDs exist
-    # as overrides are stored on personIDs.
-    # We can optimise by not going down this path when person_id doesn't exist, or
-    # no flags have experience continuity enabled
     return _get_all_feature_flags(
         all_feature_flags,
         team_id,
         distinct_id,
-        person_id,
+        person_overrides,
         groups=groups,
         property_value_overrides=property_value_overrides,
         group_property_value_overrides=group_property_value_overrides,
     )
 
 
 def set_feature_flag_hash_key_overrides(
-    feature_flags: List[FeatureFlag], team_id: int, person_id: int, hash_key_override: str
+    cursor: CursorWrapper, team_id: int, distinct_ids: List[str], hash_key_override: str
 ) -> None:
-    existing_flag_overrides = set(
-        FeatureFlagHashKeyOverride.objects.filter(team_id=team_id, person_id=person_id).values_list(
-            "feature_flag_key", flat=True
-        )
-    )
-    new_overrides = []
-    for feature_flag in feature_flags:
-        if feature_flag.ensure_experience_continuity and feature_flag.key not in existing_flag_overrides:
-            new_overrides.append(
-                FeatureFlagHashKeyOverride(
-                    team_id=team_id,
-                    person_id=person_id,
-                    feature_flag_key=feature_flag.key,
-                    hash_key=hash_key_override,
-                )
-            )
 
-    if new_overrides:
-        # :TRICKY: regarding the ignore_conflicts parameter:
-        # This can happen if the same person is being processed by multiple workers
-        # / we got multiple requests for the same person
-        # at the same time. In this case, we can safely ignore the error.
-        # We don't want to return an error response for `/decide` just because of this.
-        FeatureFlagHashKeyOverride.objects.bulk_create(new_overrides, ignore_conflicts=True)
+    query = """
+        WITH target_person_ids AS (
+            SELECT person_id FROM posthog_persondistinctid WHERE team_id = %(team_id)s AND distinct_id IN %(distinct_ids)s
+        ),
+        existing_overrides AS (
+            SELECT team_id, person_id, feature_flag_key, hash_key FROM posthog_featureflaghashkeyoverride
+            WHERE team_id = %(team_id)s AND person_id IN (SELECT person_id FROM target_person_ids)
+        ),
+        flags_to_override AS (
+            SELECT key FROM posthog_featureflag WHERE team_id = %(team_id)s AND ensure_experience_continuity = TRUE AND active = TRUE AND deleted = FALSE
+            AND key NOT IN (SELECT feature_flag_key FROM existing_overrides)
+        )
+        INSERT INTO posthog_featureflaghashkeyoverride (team_id, person_id, feature_flag_key, hash_key)
```

Comment:
don't we want to be doing an `UPSERT`? Shouldn't we be overriding existing_overrides? why not?